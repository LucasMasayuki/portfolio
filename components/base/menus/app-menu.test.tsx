import { fireEvent, render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import AppMenu from './app-menu';

describe.concurrent('App Menu component', () => {
  describe('when has not anchor element', () => {
    it('should not open menu', () => {
      const onClosse = vi.fn();
      const { container } = render(
        <AppMenu anchorEl={null} onClose={onClosse}>
          <div>test</div>
        </AppMenu>
      );

      expect(container.innerHTML).toMatchSnapshot();
    });
  });

  describe('when has anchor element', () => {
    it('should open menu', () => {
      const anchorEl = document.createElement('div');
      const onClosse = vi.fn();
      const { container } = render(
        <AppMenu anchorEl={anchorEl} onClose={onClosse}>
          <div>test</div>
        </AppMenu>
      );

      expect(container.innerHTML).toMatchSnapshot();
    });
  });

  describe('when trigger close action', () => {
    it('should close menu', () => {
      const anchorEl = document.createElement('div');
      const onClosse = vi.fn();
      const { container } = render(
        <AppMenu anchorEl={anchorEl} onClose={onClosse}>
          <div>test</div>
        </AppMenu>
      );

      fireEvent.keyDown(container, {
        key: 'Escape',
        code: 'Escape',
        keyCode: 27,
        charCode: 27,
      });

      expect(container.innerHTML).toMatchSnapshot();
    });
  });
});
