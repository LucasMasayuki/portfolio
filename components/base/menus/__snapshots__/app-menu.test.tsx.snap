// Vitest Snapshot v1

exports[`App Menu component > when has anchor element > should open menu 1`] = `""`;

exports[`App Menu component > when has not anchor element > should not open menu 1`] = `""`;

exports[`App Menu component > when trigger close action > should close menu 1`] = `""`;
